### PR TITLE
amam-suthesh/fix_livechat_rerender

### PR DESCRIPTION
**Pull request by mustofa-binary**
Changes:

- remove is_livechat_interactive and LC_API in location context
- add useLivechat hooks

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/hooks/use-livechat.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                    const currentCookie = document.cookie
+                    if (currentCookie != lastCookie) {
+                        const client_information = Cookies.get('client_information', {
+                            domain,
+                        })
+                        if (client_information) {
+                            setLoggedIn(true)
+                        } else {
+                            setLoggedIn(false)
+                        }
```
Comment (AbdallaMourad) [inline src/components/hooks/use-livechat.js:47]:
```suggestion
                        setLoggedIn(!!client_information)
```

File: src/components/hooks/use-livechat.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            }
+            if (is_livechat_interactive) {
+                window.LiveChatWidget.on('ready', () => {
+                    let session_variables = {
+                        loginid: '',
+                        landing_company_shortcode: '',
+                        currency: '',
+                        residence: '',
+                        email: '',
+                    }
```
Comment (AbdallaMourad) [inline src/components/hooks/use-livechat.js:90]: can we omit this declaration and declare it [here](https://github.com/binary-com/deriv-com/pull/1351/files#diff-e5e90c60d24e7dae6576024383c266127f2d7aef9cb1717329e7e557cc98fe9fR105) instead with fallback of empty string

File: src/components/hooks/use-livechat.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                                `${first_name} ${last_name}`,
+                            )
+                        }
+                    } else {
+                        if (window.LiveChatWidget.get('chat_data')) {
+                            const chatID = window.LiveChatWidget.get('chat_data').chatId
+                            customerSDK?.deactivateChat({ chatId: chatID })
+                        }
+                        window.LiveChatWidget.call('set_customer_email', ' ')
+                        window.LiveChatWidget.call('set_customer_name', ' ')
```
Comment (AbdallaMourad) [inline src/components/hooks/use-livechat.js:131]:
```suggestion
                        window.LiveChatWidget.call('set_customer_email', '')
                        window.LiveChatWidget.call('set_customer_name', '')
```

Should these be an empty string

Comment (mustofa-binary) [reply src/components/hooks/use-livechat.js:131]: empty string will throw an error from livechat sdk, we need to pass space for emptying email and name

Comment (AbdallaMourad) [reply src/components/hooks/use-livechat.js:131]: I see, thank you for the clarification 🙏🏻

**Review by AbdallaMourad — APPROVED** [review]

**Review by suthesan-deriv — APPROVED** [review]

**Pull request merged**